### DISCUSSION TITLE
Detect if a Blade component is rendered inside a Livewire component

### DISCUSSION THREAD

**Discussion by gbalduzzi** [livewire/livewire#7918] [1. Help] [answered]
I'm currently using Livewire v3 in a Laravel 10 app.

I defined some Blade components that I need to use both inside a "plain" blade view and inside some Livewire components.

I need to render different attributes when the component is rendered a part of a Livewire component, mostly `wire:ignore` attributes (but not only). 

I currently added a `livewire` flag to my component:

```php
<x-component livewire />
```
 
And inside the component blade:

```blade.php
<div @if($livewire) wire:ignore @endif> 
 ....
 </div>
```

Is there a way to automatically detect whether we are doing a "Livewire" rendering or not?

**Comment by danie-ramdhani** [comment, marked as answer]
how about check the subclass?

```blade
<div @if(is_subclass_of(static::class, \Livewire\Component::class)) wire:ignore @endif> 
 ....
 </div>
```

**Reply by gbalduzzi** [reply]
It works like a charm, thank you!
